templates/prometheuskuberbacproxyconfig.go

```diff
@@ -0,0 +1,42 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templates
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+var KubeRBACProxyConfigMap = corev1.ConfigMap{
+	Data: map[string]string{
+		"config-file.json": `{
+			"authorization": {
+				"static": [
+					{
+						"verb":"get",
+						"resourceRequest":false,
+						"path":"/metrics"
+					},
+					{
+						"verb":"get",
+						"resourceRequest":false,
+						"path":"/federate"
+					}
+				]
+			}
+		}`,
+	},
+}
```

Comment:
This is basically a string and we have no way to force a typed structure, every single type/structure error (like a missing comma, or special char) will be invisible until run time. I would suggest doing something similar to  

```suggestion
var KubeRBACProxyConfigMap = corev1.ConfigMap{
	Data: map[string]string{
		"config-file.json": (func() string {
			var config struct {
				Authorization struct {
					Static [2]struct {
						Path            string `json:"path"`
						ResourceRequest bool   `json:"resourceRequest"`
						Verb            string `json:"verb"`
					} `json:"static"`
				} `json:"authorization"`
			}

			item := &config.Authorization.Static[0]
			item.Verb = "get"
			item.Path = "/metrics"
			item.ResourceRequest = false

			item = &config.Authorization.Static[1]
			item.Verb = "get"
			item.Path = "/federate"
			item.ResourceRequest = false

			raw, _ := json.Marshal(config)
			return string(raw)
	})()
}
```

I do admit that defining and immediately calling an anonymous function is an inconvenient, but it does bring  benefits: 
- Everything is typed
- The anonymous struct is well defined and localized to the new func
- Everything Is defined only once
- init code is very clear
- the inlined function is called only once when the operator process starts
- The Marshal operation should never go fail, as we are marshaling a known typed struct

Reply:
Updated the code.